File: 2015/day1/Php/part2/main.php
<?php
parse_str(implode('&', array_slice($argv, 1)), $_GET);
$input = $_GET['input'];
echo "Processing: {$input} \n";
$floor = 0;
$strlen = strlen( $input );
for( $i = 0; $i <= $strlen && $floor != -1; $i++ ) {
    $char = substr( $input, $i, 1 );
    if(strcmp($char, "(") === 0) {
        ++$floor;
    } elseif(strcmp($char, ")") === 0) {
        --$floor;
    }
}
echo "{$i} \n";
?>

File: 2015/day1/Lua/part2/main.lua
floor = 0
position = 0
print("Processing: " .. arg[1])
for c in arg[1]:gmatch"." do
    if(c == "(") then
        floor = floor + 1
    else 
        floor = floor - 1
    end
    position = position + 1;
    if floor == -1 then break end
end
print(position)